thorlcr/slave/thslavemain.cpp

```diff
@@ -285,19 +283,58 @@ bool UnregisterSelf(IException *e)
     return false;
 }
 
+void signalInFuture(int signo, unsigned timeout)
+{
+#ifndef _WIN32
+    int ret;
+    timer_t timerId;
+    struct sigevent sigev;
+    struct itimerspec itSpec;
+
+    sigev.sigev_notify = SIGEV_SIGNAL;
+    sigev.sigev_signo = signo;
+    sigev.sigev_value.sival_ptr = &timerId;
+    sigev.sigev_notify_function = nullptr;
+    sigev.sigev_notify_attributes = NULL;
+
+    itSpec.it_value.tv_sec = timeout;
+    itSpec.it_value.tv_nsec = 0;
+    itSpec.it_interval.tv_sec = 0;
+    itSpec.it_interval.tv_nsec = 0;
+
+    ret = timer_create(CLOCK_MONOTONIC, &sigev, &timerId);
+    if (!ret)
+        timer_settime(timerId, 0, &itSpec, 0);
+#endif
+}
+
 bool ControlHandler(ahType type)
 {
+    static bool recvdSig = false;
+    if (recvdSig)
+    {
+        if (ahInterrupt == type)
+            _exit(128+SIGINT);
+        else
+            _exit(128+SIGTERM);
```

Comment:
is there any significance to the 128+ ?

Reply:
Standard shell exit values are 128 + signal